### PR TITLE
RSDK-3560 Add GetLatestMapInfo to SLAM Proto

### PR DESCRIPTION
**Pull request by PStrutz**
JIRA ticket: https://viam.atlassian.net/browse/RSDK-3560.

### CODE REVIEW

File: proto/viam/service/slam/v1/slam.proto [hunk 1]
```diff
@@ -85,7 +93,7 @@ message GetPointCloudMapResponse {
 }
 
 message GetInternalStateRequest {
-  // Name of slam service
```
Comment (EshaMaharishi) [inline proto/viam/service/slam/v1/slam.proto:88]: Why make this plural?

File: proto/viam/service/slam/v1/slam.proto [hunk 2]
```diff
@@ -94,3 +102,13 @@ message GetInternalStateResponse {
   // mapping/localization
   bytes internal_state_chunk = 1;
 }
+
+message GetLatestMapInfoRequest {
+  // Name of the SLAM algo
+  string name = 1;
```
Comment (EshaMaharishi) [inline proto/viam/service/slam/v1/slam.proto:108]: Will this be different from what's returned in GetInternalState?

**Review by EshaMaharishi — COMMENTED** [review]
Left a couple questions, but in general feel free to just have Jeremy review PR's for this project!

File: proto/viam/service/slam/v1/slam.proto (hunk 2 above)
Comment (PStrutz) [reply proto/viam/service/slam/v1/slam.proto:108]: Yep! From what I understand, GetInternalState retrieves all the information needed to reconstruct the point cloud (e.g. how many different chunks make up the PC, etc.) - although I am not 100% certain what EXACTLY this function returns, as this was already in the code before I worked on it. GetLatestMapInfoRequest is the new function I added to track the timestamp of the last map that was created.

Comment (EshaMaharishi) [reply proto/viam/service/slam/v1/slam.proto:108]: Ah, I misread this and thought it was being returned in the response. It makes sense to have it in the request!

**Review by jeremyrhyde — APPROVED** [review]
LGTM. Cleanly implemented!

**Pull request closed without merging**